src/torque_tilt.c

```diff
@@ -60,7 +61,37 @@ void torque_tilt_update(TorqueTilt *tt, const MotorData *motor, const RefloatCon
         step_size /= 2;
     }
 
-    rate_limitf(&tt->offset, target_offset, step_size);
+    // Smoothen changes in tilt angle by ramping the step size
+    if (config->inputtilt_smoothing_factor > 0) {
+        float smoothing_factor = 0.04;
+        // Sets the angle away from Target that step size begins ramping down
+        float smooth_center_window = 1.5;
+        float tiltback_target_diff = target_offset - tt->offset;
+
+        // Within X degrees of Target Angle, start ramping down step size
+        if (fabsf(tiltback_target_diff) < smooth_center_window) {
+            // Target step size is reduced the closer to center you are (needed for smoothly
+            // transitioning away from center)
+            tt->ramped_step_size = (smoothing_factor * step_size * (tiltback_target_diff / 2)) +
+                ((1 - smoothing_factor) * tt->ramped_step_size);
+            // Linearly ramped down step size is provided as minimum to prevent overshoot
+            float centering_step_size =
+                fminf(fabsf(tt->ramped_step_size), fabsf(tiltback_target_diff / 2) * step_size) *
+                sign(tiltback_target_diff);
+            if (fabsf(tiltback_target_diff) < fabsf(centering_step_size)) {
+                tt->offset = target_offset;
+            } else {
+                tt->offset += centering_step_size;
+            }
+        } else {
+            // Ramp up step size until the configured tilt speed is reached
+            tt->ramped_step_size = (smoothing_factor * step_size * sign(tiltback_target_diff)) +
+                ((1 - smoothing_factor) * tt->ramped_step_size);
+            tt->offset += tt->ramped_step_size;
+        }
+    } else {
+        rate_limitf(&tt->offset, target_offset, step_size);
+    }
```

Comment:
Just having a very quick look now, I'll want this extracted into a function, not copy-pasted. I think there are also multiple ways to do the smoothing, there might be a better way that'd allow simpler (needing less context) implementation in a function. At least those were the thoughts I had when looking at the original PR...